runtime/grid_reduction.cu

```diff
@@ -646,12 +646,24 @@ __device__ void serialReductionStep(
   if (!write_pred) {
     return;
   }
-  out = read_pred ? in : init;
+  if (read_pred) {
+    loadGeneric<T, vec_size>(out, in);
```

Comment:
Out of curiosity, does it matter to use `loadGeneric` here? Both `out` and `in` are registers, so I'd assume just a normal for-loop would be the same.

Reply:
Yeah for loop would work fine too.